taxi/tests.py

```diff
@@ -1,3 +1,168 @@
-from django.test import TestCase
+from django.contrib.auth import get_user_model
+from django.test import TestCase, Client
+from django.urls import reverse
 
-# Create your tests here.
+from taxi.forms import CarForm, DriverLicenseUpdateForm
+from taxi.models import Manufacturer, Driver, Car
+
+
+class ModelTests(TestCase):
+    def setUp(self):
+        self.manufacturer = Manufacturer.objects.create(
+            name="manufacturer",
+            country="country"
+        )
+        self.driver = get_user_model().objects.create(
+            username="driver",
+            password="password",
+            license_number="license_number"
+        )
+        self.car = Car.objects.create(
+            model="model",
+            manufacturer=self.manufacturer,
+        )
+        self.car.drivers.add(self.driver)
+
+    def test_string(self):
+        self.assertEqual(
+            str(self.manufacturer),
+            f"{self.manufacturer.name} {self.manufacturer.country}"
+        )
+        self.assertEqual(
+            str(self.driver),
+            f"{self.driver.username} "
+            f"({self.driver.first_name} "
+            f"{self.driver.last_name})"
+        )
+        self.assertEqual(
+            str(self.car),
+            self.car.model
+        )
+
+    def test_car_delete(self):
+        self.manufacturer.delete()
+        self.assertEqual(
+            list(Car.objects.filter(manufacturer__name="manufacturer")),
+            []
+        )
+
+
+class TestAdminRequired(TestCase):
+    def setUp(self) -> None:
+        self.admin_user = get_user_model().objects.create_superuser(
+            username="admin",
+            password="admin"
+        )
+
+    def test_without_login(self):
+        url = reverse("taxi:index")
+        res = self.client.get(url)
+        self.assertEqual(res.status_code, 302)
+
+    def with_login(self):
+        self.client.force_login(self.admin_user)
+        url = reverse("taxi:index")
+        res = self.client.get(url)
+        self.assertEqual(res.status_code, 200)
```

Comment:
The method `with_login` should be renamed to `test_with_login` to ensure it is recognized and executed as a test case by the test runner.